packages/integration-tests/src/ui-helpers/expect-page.ts

```diff
@@ -122,15 +123,29 @@ export default class ExpectPage {
     expect(this.page.url()).toBe(typeof url === 'string' ? url : url.href);
   }
 
+  /**
+   * Navigate to the given URL and wait for the page to be navigated.
+   */
+  async navigateTo(url: URL | string) {
+    const [result] = await Promise.all([
+      this.page.goto(typeof url === 'string' ? url : url.href),
+      this.page.waitForNavigation({ waitUntil: 'networkidle0' }),
+    ]);
```

Comment:
I don't think this should be wrapped in `Promise.all`. IMO, we should await `page.goto` and then `waitForNavigation`

Reply:
this is the [official way](https://pptr.dev/api/puppeteer.page.waitfornavigation#example)